buffer_period_test.go

```diff
@@ -137,4 +137,27 @@ func TestBufferPeriod(t *testing.T) {
 		bufferPeriods.Add(time.Second, 2*time.Second, "unused")
 		bufferPeriods.Stop()
 	})
+
+	t.Run("reset", func(t *testing.T) {
+		t.Parallel()
+
+		triggerCh := make(chan string, 5)
+		bufferPeriods := newTimers()
+		go bufferPeriods.Run(triggerCh)
+		defer bufferPeriods.Stop()
+
+		id := "foo"
+		bufferPeriods.Add(time.Second, 4*time.Second, id)
```

Comment:
TLDR; Not a blocker.

I've been playing with changing all the buffer_period tests to use milliseconds instead of seconds to speed up the tests. All the existing tests work fine except for 2 which just needed a small bump to expected durations. I've found that in general milliseconds work fine for most automated timing tests as they are still a pretty large unit for a computer. Maybe consider that here?

Reply:
Can do, I'll update the existing ones as well! thanks for the review